### PR TITLE
network: dhcp6pd: set lifetime and route metric

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #21000.

### CODE REVIEW

File: src/libsystemd-network/sd-dhcp6-client.c [hunk 1]
```diff
@@ -1417,6 +1427,17 @@ static int client_receive_message(
 
         sd_dhcp6_client *client = userdata;
         DHCP6_CLIENT_DONT_DESTROY(client);
+        /* This needs to be initialized with zero. See #20741. */
```
Comment (bluca) [inline src/libsystemd-network/sd-dhcp6-client.c:1430]: can the send/recv helpers in src/shared/socket-util.c be used to reduce duplication?

**Comment by yuwata** [comment]
> is it possible to cover this cases from the unit tests?

Yes. Tested in #21000.

File: src/libsystemd-network/sd-dhcp6-client.c (hunk 1 above)
Comment (yuwata) [reply src/libsystemd-network/sd-dhcp6-client.c:1430]: If possible, then that would be nice. But I do not have any good idea now.

Comment (bluca) [reply src/libsystemd-network/sd-dhcp6-client.c:1430]: never mind, that's about sending FDs and data, there's no FD here

**Pull request merged**